### ISSUE TITLE
Support for older Android devices running Android version 9 or below

### ISSUE THREAD

**Issue by Sanmeet007** [Sanmeet007/logger#3]
> dropped support for older Android devices running Android version 9 or below

Oof. My repo especially supports longevity/sustainability, e.g. wants to help those running older devices to be able having them useful for as long as possible. So that's a bit unfortunate here. And what's even more troublesome:

> it increases the application size from 20MB to 43MB

Which makes it ineligible for my repo then, which has a per-app size limit of 30 MB :cry:

So what shall we do here now? IMHO raising minSdk was not needed to get rid of the storage permissions. SAF is supported since Android 5 – and as long as you don't need any special locations (Downloads, media storage i.e. the dedicated spaces for photos and videos) no storage permissions would be needed AFAIK. So maybe you revert that minSdk change and see if it works?

If that's not possible, the only other alternative would be to use ABI specific builds (i.e. one APK per ABI). It's usually the ABI/arch specific Flutter libs blowing up file size (in your case they make up ~14 MB per ABI – so sticking to 1 ABI would reduce the APK size by ~25 MB). If you want to go that path, it would be a good idea to adjust Flutter's standard behavior concerning `versionCode`: Flutter uses `versionCode + ABI * 1000` by default (which would result in 1001, 2001, 3001 etc. for `versionCode 1`). It would be better then to change that to `versionCode * 10 + ABI` from the start (as you cannot decrease `versionCode` later).

PS: I've just tested. The armv8-only APK would be around 17 MB.

_Originally posted by @IzzySoft in https://github.com/Sanmeet007/logger/issues/1#issuecomment-2067663340_

**Comment by Sanmeet007** [comment]
@IzzySoft Ok i think we can fix it using legacy storage permission with `maxsdkversion`.

**Comment by IzzySoft** [comment]
Great! Thanks for digging that up (why didn't **I** think of that?) :hugs: And APK size is back to "normal" then?

(still strange why app size doubles if you *drop* something – maybe one time someone explains that to me, but until then I don't get it)

**Comment by Sanmeet007** [comment]
@IzzySoft  Re-supported Android versions 5 and below! Check out the latest version [v1.4.6](https://github.com/Sanmeet007/logger/releases/tag/v1.4.6). Now, let's kick back, relax, and enjoy life!

**Comment by Sanmeet007** [comment]
And to address your curiosity Below API level 23, Flutter employs uncompressed native libraries during the build process. However, above API level 23, Flutter utilizes uncompressed libraries directly. This results in the APK size being smaller than the installed size. In summary, while the APK size may differ from the installed size below API level 23, it remains the same above this threshold due to Flutter's ability to utilize uncompressed libraries directly.

For complete explanation check this stack answer : 
https://stackoverflow.com/questions/62440105/apk-size-increased-35mb-when-bumping-min-sdk-from-21-to-24/63417612#63417612

**Comment by IzzySoft** [comment]
> Waiting for your repo to update build version

Yay! Updater kicks in in less than an hour from now. Good timing! And size is back to normal.

Btw, the "funny size question": Once you moved up minSdk, the libs were no longer "compressed" but simply "stored". That's what caused the size increase. But don't ask me why they decided that. Background of not compressing is to not having to extract them (newer Android versions can use the libs directly from the APK *if they are not compressed*). To save storage space. Counter-productive with apps built using Flutter or RN if they contain multiple ABIs, but there's no if-then-else here.

Ah, I see your second comment (I was distracted after starting mine, so it crossed) already has that detail, you found that already…

**Comment by Sanmeet007** [comment]
Glad to hear, and wow, you're a curious being!

**Comment by IzzySoft** [comment]
> wow, you're a curious being!

I like to learn :wink: I've learned a lot on these things since I started with my repo about 8 years ago. Still far too many gaps in my knowledge. If I want to be able helping the developers with their apps on my repo, as well as those using them, and as well to make it a place both can enjoy and feel safe with – well, I have to be curious to fill those gaps. At least as far as I can. I can't go to the length of becoming a developer myself (after all, it's not a "paid job" I do with this stuff) – but I'll certainly do my best where I can.

So I take this as a compliment – and thanks! :star_struck:

**Comment by Sanmeet007** [comment]
@IzzySoft  Hey man ! After reading documentation and testing the app on lower android versions .. there is no need for requesting legacy storage or any of these permission : `read_external_storage` and `write_external_storage`.

The code i wrote uses temporary directories so no need to get permission reading that and about write external storage ... its handled by SAF ! 

Finally fixed the app for final time => new version [v1.4.7](https://github.com/Sanmeet007/logger/releases/tag/v1.4.7)


PS: Please remove the  WRITE_EXTERNAL_STORAGE & READ_EXTERNAL_STORAGE permissions from the permissions section in your repo !

**Comment by IzzySoft** [comment]
> After reading documentation and testing the app on lower android versions .. there is no need for requesting legacy storage or any of these permission

That's what I assumed, which is why I initially wrote that. Thanks for checking out and confirming, I could have been wrong there after all…

> its handled by SAF

Yupp. So no need for a broad permission as those using the app have "fine-grained control" :smiley: 

> Please remove the WRITE_EXTERNAL_STORAGE & READ_EXTERNAL_STORAGE permissions from the permissions section in your repo !

Once a new release comes in no longer requesting them, my updater will alert me about that and I'll remove the corresponding lines from the app's "green list". I cannot remove permissions "from the permissions section" as that's handled automatically: that section simply shows the permissions requested in the Manifest of the corresponding APK :man_shrugging: (I'd go nuts had I to do that manually for more than 1k apps in the repo :see_no_evil:)

**TL;DR:** your next release will take care for that automatically.

**Comment by Sanmeet007** [comment]
@IzzySoft  I learnt a lot from your advice thanks for improving me 😊

**Comment by IzzySoft** [comment]
Thanks for walking that path with me! I really enjoyed it.
